File: javaFiles/max.java
public class max {
    public static void main(String[] args){
        System.out.println(max(
                new int[] {-1, 0, 2,5,9}
        ));
    }

    public static int max (int[] a){
        int result = a[0];
        for(int i = 1; i < a.length; i++){
            if(a[i] > result){
                result = a[i];
            }
        }

        return result;
    }
}

File: max/max.js
class MAX {
	constructor(a=[]){
		this.a = a;
		this.i = 1;
		this.result = a[0];
		this.node1 = "";
	}

	update(){

		if(this.i < this.a.length){
			if(this.a[this.i] > this.result){
				this.result = this.a[this.i];
				this.node1 = "A pointer fölött álló szám nagyobb, mint a jelenlegi állás, ezért a pointer fölött álló elemet belerakjuk a jelenlegi állásba és tövábbléptetünk.";
			}else{
				this.node1 = "A pointer fölött álló szám nem nagyobb, mint a jelenlegi állás, ezért továbblépünk."

			}
			this.i++;
		}
	}

	show () {
		return {
			pointerA: this.i,
			currentResult: [this.result],
			comparison: this.node1
		}

	}
}
